app/src/main/java/com/github/se/project/ui/lesson/SelectedTutorDetails.kt

```diff
@@ -0,0 +1,379 @@
+            }
+      }
+}
+
+@Composable
+private fun TutorInfoSection(profile: Profile) {
+  Row(
+      modifier = Modifier.fillMaxWidth().testTag("tutorInfoRow"),
+      horizontalArrangement = Arrangement.spacedBy(16.dp),
+      verticalAlignment = Alignment.CenterVertically) {
+        // Profile picture placeholder
+        DisplayProfilePicture("tutorProfilePicture")
+
+        // Tutor details
+        Column(modifier = Modifier.weight(1f)) {
+          Text(
+              text = "${profile.firstName} ${profile.lastName}",
+              style = MaterialTheme.typography.titleMedium,
+              modifier = Modifier.testTag("tutorName"))
+          Text(
+              text = "${profile.section} - ${profile.academicLevel}",
+              style = MaterialTheme.typography.bodyMedium,
+              color = MaterialTheme.colorScheme.onSurfaceVariant,
+              modifier = Modifier.testTag("tutorAcademicInfo"))
+        }
+
+        // Price and rating
+        Column(horizontalAlignment = Alignment.End) {
+          DisplayRatingGrade(profile.rating.averageRating, "tutorRating")
+
+          Text(
+              text = "${profile.price}.-/h",
+              style = MaterialTheme.typography.labelMedium,
+              color = MaterialTheme.colorScheme.primary,
+              modifier = Modifier.testTag("tutorPrice"))
+        }
+      }
+}
+
+@Composable
+private fun TutorDescriptionSection(description: String) {
+  Column(
+      verticalArrangement = Arrangement.spacedBy(8.dp),
+      modifier = Modifier.fillMaxWidth().testTag("tutorDescriptionSection")) {
+        if (description.isEmpty()) {
+          Text(
+              text = "No description available.",
+              style = MaterialTheme.typography.bodyMedium,
+              modifier = Modifier.testTag("tutorDescriptionEmpty"))
+        } else {
+          Text(
+              text = description,
+              style = MaterialTheme.typography.bodyMedium,
+              modifier = Modifier.testTag("tutorDescription"))
+        }
+      }
+}
+
+@Composable
+private fun TutorCommentsSection(
+    comments: MutableList<Comment>,
+    listProfilesViewModel: ListProfilesViewModel
+) {
+  Column(
+      verticalArrangement = Arrangement.spacedBy(8.dp),
+      modifier = Modifier.fillMaxWidth().testTag("tutorCommentsSection")) {
+        Text(
+            text = "Best reviews on this tutor",
+            style = MaterialTheme.typography.titleMedium,
+            modifier = Modifier.testTag("tutorCommentsTitle"))
+
+        if (comments.isEmpty()) {
+          Text(
+              text = "No reviews available.",
+              style = MaterialTheme.typography.bodyMedium,
+              modifier = Modifier.testTag("tutorCommentsEmpty"))
+        } else {
+          comments.sortByDescending { it.grade }
+          comments.take(3).forEach {
+            val student = listProfilesViewModel.getProfileById(it.raterUid)
+            if (student != null) {
```

Comment:
Very smart to check for potentially deleted accounts. I think they should maybe still be displayed under "deleted user", but this also works